dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java

```diff
@@ -1,343 +1,200 @@
-						"Locking of db_version table failed: "
-								+ e2.getMessage());
-				throw new DotRuntimeException(
-						"Locking of db_version table failed: "
-								+ e2.getMessage(), e2);
-			}
 
+			});
+		} catch (DotDataException e) {
+			LocalTransaction.wrap(()->{
+				Config.DB_VERSION = 0;
+				new DotConnect().setSQL(create).loadResult(); 
+			});
+			
 		}
-		Logger.debug(this.getClass(), "Locking db_version succeeded");
-
-		boolean firstTimeStart = false;
+	}
+	
+	Comparator<Class<?>> comparator = new Comparator<Class<?>>() {
+		public int compare(Class<?> o1, Class<?> o2) {
+			return o1.getName().compareTo(o2.getName());
+		}
+	};
+	
+	
+	public void executeStartup() throws DotDataException, DotRuntimeException {
+		setupDBVersionTable();
+		List<Class<?>> runAlways= TaskLocatorUtil.getStartupRunAlwaysTaskClasses();
+		Collections.sort(runAlways, comparator);
 		String name = null;
 		try {
 			Logger.info(this, "Starting startup tasks.");
-			HibernateUtil.startTransaction();
-
-			for (Class<?> c : runAlways) {
+			for (final Class<?> c : runAlways) {
 				name = c.getCanonicalName();
 				name = name.substring(name.lastIndexOf(".") + 1);
 				if (StartupTask.class.isAssignableFrom(c)) {
-					StartupTask task;
-					try {
-						task = (StartupTask) c.newInstance();
-					} catch (Exception e) {
-						throw new DotRuntimeException(e.getMessage(), e);
-					}
-					HibernateUtil.startTransaction();
-					if (task.forceRun()) {
-						HibernateUtil.commitTransaction();
-						HibernateUtil.startTransaction();
-						Logger.info(this, "Running: " + name);
-						task.executeUpgrade();
-						if(name.equals("Task00001LoadSchema")){
-							firstTimeStart = true;
+					
+					LocalTransaction.wrap(()->{
+
+						StartupTask task;
+						try {
+							task = (StartupTask) c.newInstance();
+						} catch (Exception e) {
+							throw new DotRuntimeException(e.getMessage(), e);
 						}
-					} else {
-						Logger.info(this, "Not running: " + name);
-					}
-					HibernateUtil.commitTransaction();
+
+						if (task.forceRun()) {
+							Logger.info(this, "Running: " + c.getName());
+							task.executeUpgrade();
+							if(c.getName().equals(Task00001LoadSchema.class.getName())){
+								System.setProperty(DOTCMS_FIRST_TIME_START, "true");
+							}
+						} else {
+							Logger.info(this, "Not running: " + c.getName());
+						}
+					});
+
 				}
 			}
 			Logger.info(this, "Finishing startup tasks.");
 		} catch (Exception e) {
-			HibernateUtil.rollbackTransaction();
 			Logger.fatal(this, "Unable to execute the upgrade task : " + name, e);
-		} finally {
-			// This will commit the changes and close the connection
-			HibernateUtil.closeSession();
-			try {
-			    update.close();
-//			    conn.close();
-			} catch(Exception exx) { /* Ignore this */ }
+			throw new DotDataException("Unable to execute startup task : ",e);
 		}
+	}
+	
+	
+	public void executeUpgrades() throws DotDataException, DotRuntimeException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.exception.DotRuntimeException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)